packages/js/components/src/experimental-select-tree-control/select-tree.tsx

```diff
@@ -159,40 +205,116 @@ export const SelectTree = function SelectTree( {
+					shouldShowCreateButton?.( props.createValue )
+				) {
+					setHighlightedIndex( items.length );
+				} else {
+					const visibleNodeIndex = getVisibleNodeIndex(
+						linkedTree,
+						Math.min( highlightedIndex + 1, items.length ),
+						'down'
+					);
+					setHighlightedIndex(
+						visibleNodeIndex !== undefined
+							? visibleNodeIndex
+							: highlightedIndex
+					);
+				}
+			} else if ( event.key === 'ArrowUp' ) {
+				if ( highlightedIndex > 0 ) {
+					const visibleNodeIndex = getVisibleNodeIndex(
+						linkedTree,
+						Math.max( highlightedIndex - 1, -1 ),
+						'up'
+					);
+					setHighlightedIndex(
+						visibleNodeIndex !== undefined
+							? visibleNodeIndex
+							: highlightedIndex
+					);
+				} else {
+					setHighlightedIndex( -1 );
+				}
 			} else if ( event.key === 'Tab' || event.key === 'Escape' ) {
 				setIsOpen( false );
 				recalculateInputValue();
-			} else if ( event.key === ',' || event.key === 'Enter' ) {
+			} else if ( event.key === 'Enter' || event.key === ',' ) {
 				event.preventDefault();
-				const item = items.find(
-					( i ) => i.label === escapeHTML( inputValue )
-				);
-				const isAlreadySelected =
-					Array.isArray( props.selected ) &&
-					Boolean(
-						props.selected.find(
-							( i ) => i.label === escapeHTML( inputValue )
-						)
+				if (
+					highlightedIndex === items.length &&
+					shouldShowCreateButton
+				) {
+					props.onCreateNew?.();
+				} else if (
+					// is selecting an item
+					highlightedIndex !== -1
+				) {
+					if ( props.multiple && Array.isArray( props.selected ) ) {
+						if (
+							! Boolean(
+								props.selected.find(
+									( i ) =>
+										i.label ===
+										items[ highlightedIndex ].label
+								)
+							)
+						) {
+							props.onSelect?.( items[ highlightedIndex ] );
+						} else {
+							props.onRemove?.( items[ highlightedIndex ] );
+						}
+						setInputValue( '' );
+					} else {
+						onInputChange?.( items[ highlightedIndex ].label );
+						setIsOpen( false );
+						setIsFocused( false );
+						focusOnInput();
+					}
+				} else if ( inputValue ) {
+					// no highlighted item, but there is an input value, check if it matches any item
+
+					const item = items.find(
+						( i ) => i.label === escapeHTML( inputValue )
 					);
-				if ( props.onSelect && item && ! isAlreadySelected ) {
-					props.onSelect( item );
-					setInputValue( '' );
-					recalculateInputValue();
+					const isAlreadySelected = Array.isArray( props.selected )
+						? Boolean(
+								props.selected.find(
+									( i ) =>
+										i.label === escapeHTML( inputValue )
+								)
+						  )
+						: props.selected?.label === escapeHTML( inputValue );
+					if ( item && ! isAlreadySelected ) {
+						props.onSelect?.( item );
+						setInputValue( '' );
+						recalculateInputValue();
+					}
 				}
 			} else if (
-				( event.key === 'ArrowLeft' || event.key === 'Backspace' ) &&
```

Comment:
Remove the arrow left key from moving to the token items, since now the user could be wanting to collapse a node.